src/schemas/ev42/ev42_rw.cxx

```diff
@@ -47,70 +79,54 @@ class writer : public FBSchemaWriter {
   uint64_t ts_max = 0;
 };
 
-std::unique_ptr<FBSchemaWriter> reader::create_writer_impl() {
-  return std::unique_ptr<FBSchemaWriter>(new writer);
+FileWriter::HDFWriterModule::ptr HDFWriterModule::create() {
+  return FileWriter::HDFWriterModule::ptr(new HDFWriterModule);
 }
 
-static EventMessage const *get_fbuf(char const *data) {
-  return GetEventMessage(data);
-}
-
-bool reader::verify_impl(Msg msg) {
-  auto veri = flatbuffers::Verifier((uint8_t *)msg.data, msg.size);
-  if (VerifyEventMessageBuffer(veri))
-    return true;
-  return false;
-}
-
-uint64_t reader::ts_impl(Msg msg) {
-  auto fbuf = get_fbuf(msg.data);
-  return fbuf->pulse_time();
-}
-
-std::string reader::sourcename_impl(Msg msg) {
-  auto fbuf = get_fbuf(msg.data);
-  auto v = fbuf->source_name();
-  if (!v) {
-    LOG(4, "WARNING message has no source name");
-    return "";
+HDFWriterModule::InitResult
+HDFWriterModule::init_hdf(hid_t hdf_file, std::string hdf_parent_name,
+                          rapidjson::Value const &config_stream,
+                          rapidjson::Value const *config_module) {
```

Comment:
Not yet.

Reply:
I suggest adding things only when they are being used otherwise the plan changes and you are left with half implemented bits all over the codebase. (YAGNI)